app/scripts/lib/metaRPCClientFactory.test.js

```diff
@@ -131,4 +131,16 @@ describe('metaRPCClientFactory', () => {
       },
     });
   });
+
+  it('should be able to handle no message within TIMEOUT secs', async () => {
+    const streamTest = createThoughStream();
+    const metaRPCClient = metaRPCClientFactory(streamTest);
+    metaRPCClient.foo('bad', (error, _) => {
+      expect(error.message).toStrictEqual('No response from RPC');
+    });
+
+    const delay = (ms) => new Promise((res) => setTimeout(res, ms));
+
+    await delay(11000);
```

Comment:
Can we use Jest and/or Sinon's fake timers feature to fast-forward through time instead of waiting 11 seconds?

Reply:
Using fake timers, I can only check that setTimeout was called. I can't check that the callback is called with the error.

If I use fakeTimer I get this error
```
    TypeError: cb is not a function

      21 |     setTimeout(() => {
      22 |       if (!this.responseHandled[id] && cb) {
    > 23 |         return cb(new Error('No response from RPC'), null);
         |                ^
      24 |       }
```

If I console-log cb before setTimeout, it's properly set, but with fakeTimer, by the time the body of settimeout function is being called, it's somehow no longer in scope.